authutils/auth_service.go

```diff
@@ -105,3 +109,58 @@ func VerifyAddress(address common.Address, otherAddress common.Address) error {
 	}
 	return nil
 }
+
+func SignMessageForMetering(req *http.Request) ( ) {
+
+	privateKey,err := getPrivateKeyForMetering()
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	 currentBlock,err := CurrentBlock();
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+    signature := signForMeteringValidation(privateKey,currentBlock,MeteringPrefix)
+	req.Header.Set("X-Signature",string(signature))
+	req.Header.Set("X-Currentblocknumber",currentBlock.String())
+}
+
+func getPrivateKeyForMetering()  (privateKey *ecdsa.PrivateKey,err error) {
+	if privateKeyString := config.GetString(config.PvtKeyForMetering); privateKeyString != "" {
+		privateKey, err = crypto.HexToECDSA(privateKeyString)
+		if err != nil {
+			return nil, err
+		}
+	}
+	log.Debug(crypto.PubkeyToAddress(privateKey.PublicKey))
+	return
+}
+
+func signForMeteringValidation(privateKey *ecdsa.PrivateKey, currentBlock *big.Int, prefix string) []byte {
```

Comment:
Is it possible to include the UserName or UserAccount in the Signature?
So that only the metering happens for the respective user rather than just trusting the UserName parameter to metering service.

Reply:
Also can you please check that the build is failing.

Reply:
We may even have to look at the edge scenarios, like what happens if the connectivity to metering service is lost? Do we need to have a retry mechanism or we can ignore like fire and forget?
For now we have a issue and will implement it later.

Reply:
> Is it possible to include the UserName or UserAccount in the Signature?
> So that only the metering happens for the respective user rather than just trusting the UserName parameter to metering service.

Added now

Reply:
> Also can you please check that the build is failing.

Fixed this

Reply:
> > Is it possible to include the UserName or UserAccount in the Signature?
> > So that only the metering happens for the respective user rather than just trusting the UserName parameter to metering service.
> 
> Added now

we do have a re try mechanism , but needs more thought like how many attempts or how long to wait before an other attempt is made  , for now will track this as a seprate issue as advised

Reply:
#377 https://github.com/singnet/snet-daemon/issues/377